RealmSwift/Projection.swift

```diff
@@ -0,0 +1,1342 @@
+     ```swift
+     class Person: Object {
+         @Persisted var dogs: MutableSet<Dog>
+     }
+     class PersonProjection: Projection<Person> {
+         @Projected(\Person.dogs.projectTo.name) var dogNames: ProjectedMutableSet<String>
+     }
+    ```
+     In this code the `Person`'s dogs set will be prijected to the list of dogs names via `projectTo`
+     */
+    public var projectTo: MutableSetElementMapper<Element> {
+        MutableSetElementMapper(set: self)
+    }
+}
+
+// MARK: CustomObjectiveCBridgeable
+
+// FIXME: Remove when `as! Self` can be written
+private func forceCastToInferred<T, V>(_ x: T) -> V {
+    return x as! V
+}
+
+extension Projection: CustomObjectiveCBridgeable {
+    internal static func bridging(objCValue objectiveCValue: Any) -> Self {
+        return Self(projecting: forceCastToInferred(objectiveCValue))
+    }
+
+    internal var objCValue: Any {
+        self.rootObject.unsafeCastToRLMObject()
+    }
+}
+
+#if canImport(Combine)
+// MARK: - RealmSubscribable
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, *)
+extension ProjectionObservable {
+    /// :nodoc:
+    public func _observe<S>(_ keyPaths: [String]?, on queue: DispatchQueue?, _ subscriber: S) -> NotificationToken where S: Subscriber, S.Input == Self, S.Failure == Error {
+        return observe(keyPaths: keyPaths ?? [], on: queue) { (change: ObjectChange<S.Input>) in
+            switch change {
+            case .change(let projection, _):
+                _ = subscriber.receive(projection)
+            case .deleted:
+                subscriber.receive(completion: .finished)
+            case .error(let error):
+                subscriber.receive(completion: .failure(error))
+            }
+        }
+    }
+
+    /// :nodoc:
+    public func _observe<S>(_ keyPaths: [String]?, _ subscriber: S) -> NotificationToken where S: Subscriber, S.Failure == Never, S.Input == Void {
+        return observe(keyPaths: [PartialKeyPath<Self>](), { _ in _ = subscriber.receive() })
+    }
+}
+#if !(os(iOS) && (arch(i386) || arch(arm)))
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, *)
+extension Projection: ObservableObject, RealmSubscribable where Root: ThreadConfined {
+    /// A publisher that emits Void each time the projection changes.
+    ///
+    /// Despite the name, this actually emits *after* the projection has changed.
+    public var objectWillChange: RealmPublishers.WillChange<Projection> {
+        RealmPublishers.WillChange(self)
+    }
+}
+#endif // !(os(iOS) && (arch(i386) || arch(arm)))
+
+#endif // canImport(Combine)
+
+// MARK: Projected Mutable Set
+
+/// ProjectedMutableSet is a special type of collection for Projection's properties
+/// You don't need to instantiate this type manually.
+public struct ProjectedMutableSet<NewElement>: RandomAccessCollection where NewElement: RealmCollectionValue {
```

Comment:
Is this actually a different type from ProjectedList? At first glance it appears that both could be ProjectedCollection.

Reply:
Interface similar, but isn't matching, like other our collections